curvefs/src/client/base_client.h

```diff
@@ -0,0 +1,175 @@
+                            brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void CreateDentry(const Dentry &dentry,
+                              CreateDentryResponse *response,
+                              brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void DeleteDentry(uint32_t fsId, uint64_t inodeid,
+                              const std::string &name,
+                              DeleteDentryResponse *response,
+                              brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void GetInode(uint32_t fsId, uint64_t inodeid,
+                          GetInodeResponse *response, brpc::Controller *cntl,
+                          brpc::Channel *channel);
+
+    virtual void UpdateInode(const Inode &inode, UpdateInodeResponse *response,
+                             brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void CreateInode(const InodeParam &param,
+                             CreateInodeResponse *response,
+                             brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void DeleteInode(uint32_t fsId, uint64_t inodeid,
+                             DeleteInodeResponse *response,
+                             brpc::Controller *cntl, brpc::Channel *channel);
+};
+
+class MDSBaseClient {
+ public:
+    virtual void CreateFs(const std::string &fsName, uint64_t blockSize,
+                          const Volume &volume, CreateFsResponse *response,
+                          brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void DeleteFs(const std::string &fsName, DeleteFsResponse *response,
+                          brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void MountFs(const std::string &fsName, const MountPoint &mountPt,
+                         MountFsResponse *response, brpc::Controller *cntl,
+                         brpc::Channel *channel);
+
+    virtual void UmountFs(const std::string &fsName, const MountPoint &mountPt,
+                          UmountFsResponse *response, brpc::Controller *cntl,
+                          brpc::Channel *channel);
+
+    virtual void GetFsInfo(const std::string &fsName,
+                           GetFsInfoResponse *response, brpc::Controller *cntl,
+                           brpc::Channel *channel);
+
+    virtual void GetFsInfo(uint32_t fsId, GetFsInfoResponse *response,
+                           brpc::Controller *cntl, brpc::Channel *channel);
+};
+
+class SpaceBaseClient {
+ public:
+    virtual void AllocExtents(uint32_t fsId,
+                              const ExtentAllocInfo &toAllocExtent,
+                              curvefs::space::AllocateType type,
+                              AllocateSpaceResponse *response,
+                              brpc::Controller *cntl, brpc::Channel *channel);
+
+    virtual void DeAllocExtents(uint32_t fsId,
+                                std::list<Extent> allocatedExtents,
```

Comment:
use "const std::list<Extent> &allocatedExtents" instead to avoid copy

Reply:
done